### ISSUE TITLE
:cineast-core:generateProto doesn't output java files on new Debian bullseye with OpenJDK 14

### ISSUE THREAD

**Issue by frankier** [vitrivr/cineast#111]
This is an issue that occurs on my own branch, which is just dev + https://github.com/vitrivr/cineast/pull/110

The machine has a fresh Debian bulleye install with OpenJDK 14

When I run:

    $ ./gradlew :cineast-core:generateProto

I don't get any .java files generated, only .proto!

Oddly it does work in the following situations:

 * Debian bullseye upgraded from buster with openjdk 11.0.7 (my old laptop)
 * Debian buster with openjdk:14-buster (Singularity environment based on Docker image build see https://singularity-hub.org/containers/13554/log )

I'm left wondering whether I might be missing some hidden system dependency. I tried installing the various protobuf packages from apt just in case, but it hasn't helped.

**Comment by frankier** [comment]
For completeness, here is a directory tree on my old laptop:

```
~/e/r/cineast ❯❯❯ tree cineast-core/build                            gsoc2020 ◼
cineast-core/build
├── extracted-include-protos
│   └── main
│       └── google
│           ├── api
│           │   ├── annotations.proto
│           │   ├── auth.proto
│           │   ├── backend.proto
│           │   ├── billing.proto
│           │   ├── config_change.proto
│           │   ├── consumer.proto
│           │   ├── context.proto
│           │   ├── control.proto
│           │   ├── distribution.proto
│           │   ├── documentation.proto
│           │   ├── endpoint.proto
│           │   ├── experimental
│           │   │   ├── authorization_config.proto
│           │   │   └── experimental.proto
│           │   ├── httpbody.proto
│           │   ├── http.proto
│           │   ├── label.proto
│           │   ├── logging.proto
│           │   ├── log.proto
│           │   ├── metric.proto
│           │   ├── monitored_resource.proto
│           │   ├── monitoring.proto
│           │   ├── quota.proto
│           │   ├── service.proto
│           │   ├── source_info.proto
│           │   ├── system_parameter.proto
│           │   └── usage.proto
│           ├── cloud
│           │   └── audit
│           │       └── audit_log.proto
│           ├── logging
│           │   └── type
│           │       ├── http_request.proto
│           │       └── log_severity.proto
│           ├── longrunning
│           │   └── operations.proto
│           ├── protobuf
│           │   ├── any.proto
│           │   ├── api.proto
│           │   ├── compiler
│           │   │   └── plugin.proto
│           │   ├── descriptor.proto
│           │   ├── duration.proto
│           │   ├── empty.proto
│           │   ├── field_mask.proto
│           │   ├── source_context.proto
│           │   ├── struct.proto
│           │   ├── timestamp.proto
│           │   ├── type.proto
│           │   └── wrappers.proto
│           ├── rpc
│           │   ├── code.proto
│           │   ├── error_details.proto
│           │   └── status.proto
│           └── type
│               ├── color.proto
│               ├── date.proto
│               ├── dayofweek.proto
│               ├── latlng.proto
│               ├── money.proto
│               ├── postal_address.proto
│               └── timeofday.proto
├── extracted-protos
│   └── main
└── generated
    └── main
        ├── grpc
        │   └── org
        │       └── vitrivr
        │           ├── adampro
        │           │   └── grpc
        │           │       ├── AdamDefinitionGrpc.java
        │           │       └── AdamSearchGrpc.java
        │           └── cottontail
        │               └── grpc
        │                   ├── CottonDDLGrpc.java
        │                   ├── CottonDMLGrpc.java
        │                   └── CottonDQLGrpc.java
        └── java
            └── org
                └── vitrivr
                    ├── adampro
                    │   └── grpc
                    │       └── AdamGrpc.java
                    └── cottontail
                        └── grpc
                            └── CottontailGrpc.java

32 directories, 59 files
```

**Comment by frankier** [comment]
And on the new laptop/fresh bulleye install>

```
garmak% tree cineast-core/build/          
cineast-core/build/
├── extracted-include-protos
│   └── main
│       └── google
│           ├── api
│           │   ├── annotations.proto
│           │   ├── auth.proto
│           │   ├── backend.proto
│           │   ├── billing.proto
│           │   ├── config_change.proto
│           │   ├── consumer.proto
│           │   ├── context.proto
│           │   ├── control.proto
│           │   ├── distribution.proto
│           │   ├── documentation.proto
│           │   ├── endpoint.proto
│           │   ├── experimental
│           │   │   ├── authorization_config.proto
│           │   │   └── experimental.proto
│           │   ├── httpbody.proto
│           │   ├── http.proto
│           │   ├── label.proto
│           │   ├── logging.proto
│           │   ├── log.proto
│           │   ├── metric.proto
│           │   ├── monitored_resource.proto
│           │   ├── monitoring.proto
│           │   ├── quota.proto
│           │   ├── service.proto
│           │   ├── source_info.proto
│           │   ├── system_parameter.proto
│           │   └── usage.proto
│           ├── cloud
│           │   └── audit
│           │       └── audit_log.proto
│           ├── logging
│           │   └── type
│           │       ├── http_request.proto
│           │       └── log_severity.proto
│           ├── longrunning
│           │   └── operations.proto
│           ├── protobuf
│           │   ├── any.proto
│           │   ├── api.proto
│           │   ├── compiler
│           │   │   └── plugin.proto
│           │   ├── descriptor.proto
│           │   ├── duration.proto
│           │   ├── empty.proto
│           │   ├── field_mask.proto
│           │   ├── source_context.proto
│           │   ├── struct.proto
│           │   ├── timestamp.proto
│           │   ├── type.proto
│           │   └── wrappers.proto
│           ├── rpc
│           │   ├── code.proto
│           │   ├── error_details.proto
│           │   └── status.proto
│           └── type
│               ├── color.proto
│               ├── date.proto
│               ├── dayofweek.proto
│               ├── latlng.proto
│               ├── money.proto
│               ├── postal_address.proto
│               └── timeofday.proto
└── extracted-protos
    └── main
```

**Comment by lucaro** [comment]
Very interesting. This does not look like a cineast-specific issue though, have you opened an issue with the protobuf or grpc projects?

**Comment by frankier** [comment]
I'm not sure, but I think this must have been a consequence of somehow only checking out one layers of git submodules rather than both layers. I could've sworn I did a recursive clone. Perhaps the problem was I did a recursive clone and then switched branches without updating submodules. I suppose it could be nice to somehow warn if the expected stuff isn't there but since this is entirely avoidable I'm closing this issue.
